common/workunit/workunit.cpp

```diff
@@ -14269,16 +14269,16 @@ void executeThorGraph(const char * graphName, IConstWorkUnit &workunit, const IP
         unsigned runningTimeLimit = workunit.getDebugValueInt("maxRunTime", 0);
         runningTimeLimit = runningTimeLimit ? runningTimeLimit : INFINITE;
 
-        std::list<WUState> expectedStates = { WUStateRunning, WUStateWait };
+        std::list<WUState> expectedStates = { WUStateRunning, WUStateWait, WUStateFailed };
```

Comment:
NB: not absolutely required in this commit, but related. This code should have handled the workunit having put in the failed state before. The only downside before was the production of an extra spurious "Query <wuid> failed, state: ..." exception